### PR TITLE
fix: function type comparison adds 'restParamIdx' comparison item

### PR DESCRIPTION
**Pull request by Shanks0224**

### CODE REVIEW

File: tests/samples/class_direct_call.ts [hunk 1]
```diff
@@ -32,3 +32,20 @@ export function foo() {
     }
     a.test();
 }
+
+class Foo {
+    method_foo(arr: number[]): number {
```
Comment (xujuntwt95329) [inline tests/samples/class_direct_call.ts:37]: The test case doesn't use rest parameters, seems not related to the modification?

Comment (Shanks0224) [reply tests/samples/class_direct_call.ts:37]: When creating a **_FunctionType_** based on a **_TSFunction_**, it will first use the `findValueTypeByType` function to find whether the **_TSFunction_** already exists in the cache `Map<Type, ValueType>`.
A function type is uniquely identified through **_TSFunction_**'s return value, parameter list, and envParamLen.

Comment (Shanks0224) [reply tests/samples/class_direct_call.ts:37]: When processing parameters in a function call, the processing logic is different whether **restParamIdx** is greater than -1.

Comment (Shanks0224) [reply tests/samples/class_direct_call.ts:37]: When processing parameters and arguments during a function call, the parameter type is calculated through the `getParamType`.
The **restParamIdx** property value of parameter will affect the return of the `getParamType`.

**Review by xujuntwt95329 — APPROVED** [review]
LGTM
